x-pack/legacy/plugins/task_manager/task_manager.ts

```diff
@@ -78,8 +89,9 @@ export class TaskManager {
       index: opts.config.get('xpack.task_manager.index'),
       maxAttempts: opts.config.get('xpack.task_manager.max_attempts'),
       definitions: this.definitions,
-      kibanaId: opts.config.get('server.uuid'),
+      kibanaId: ensureTaskManagerUUID(opts.config.get('server.uuid'), this.logger),
```

Comment:
I'm on the -- side to this change. It feels like the wrong place to address an invalid `server.uuid` in config. Plugins should boot up _after_ config. I really want to know why that's happening in the first place.

@elastic/kibana-operations thoughts?

Reply:
Agreed.
Opened an issue to track that: https://github.com/elastic/kibana/issues/47388

I'd like to push this as this currently breaks on master though.

Reply:
Considering that trying to derive the UUID from config is expected to fail at this point in server start, how about just directly generating the UUID right here?

After that change, this would LGTM.

Having a special UUID for TM should be mostly harmless, since nothing programmatic will fail if the TM UUID doesn't match the Kibana UUID.

Reply:
I can live with that.
Pushing a change :)